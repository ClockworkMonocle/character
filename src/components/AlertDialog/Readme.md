Uses the `<Modal />` component to render an "Are you sure?" dialog to the user.

**NOTE** There is no way to cancel the modal other than a user responding via yes/no.

```js
require('../Button');
require('../Type');

const toggle = () => setState({ active: !state.active });
const handleConfirm = answer => {
  toggle();
  alert(`User responded: ${answer}`);
};

<div>
  <Button color="red" onClick={toggle}>Delete</Button>
  <AlertDialog
    active={state.active}
    message={<Type>Are you sure you want to delete all of your data?</Type>}
    onConfirm={handleConfirm}
  />
</div>
```
