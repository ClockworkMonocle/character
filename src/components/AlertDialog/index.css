.alert {
  background-color: var(--white);
  border-radius: 2px;
  padding: 12px;
  max-width: 240px;
  max-height: 85vh;
}

.alert__controls {
  width: 100%;
  display: flex;
  justify-content: flex-end;
  padding-top: 24px;
}

.alert__control {
  margin-right: 24px;
}
