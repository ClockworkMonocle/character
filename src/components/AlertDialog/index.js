import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'components/Modal';
import Button from 'components/Button';
import Type from 'components/Type';

import './index.css';

/* istanbul ignore next */ const noop = () => {};

export default function AlertDialog({ active, message, onConfirm }) {
  return (
    <Modal onDismiss={noop} active={active}>
      <div className="alert">
        {message}
        <div className="alert__controls">
          <Button className="alert__control" variant="bare" color="red" onClick={/* istanbul ignore next */ () => onConfirm('no')}>No</Button>
          <Button variant="bare" color="green" onClick={/* istanbul ignore next */ () => onConfirm('yes')}>Yes</Button>
        </div>
      </div>
    </Modal>
  );
}

AlertDialog.propTypes = {
  /** Boolean to determine to show this confirmation or not */
  active: PropTypes.bool,
  /** The message on which you want confirmation */
  message: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Type)
  ]).isRequired,
  /** The callback function to be called with the response of the user: 'yes': 'no' */
  onConfirm: PropTypes.func.isRequired,
};

AlertDialog.defaultProps = {
  active: false,
};
