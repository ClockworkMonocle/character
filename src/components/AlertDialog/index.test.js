import React from 'react';
import { shallow } from 'enzyme';
import AlertDialog from './index';

test('<AlertDialog /> does not render', () => {
  expect(shallow(<AlertDialog active message="yo?" onConfirm={() => {}}/>)).toMatchSnapshot();
});
