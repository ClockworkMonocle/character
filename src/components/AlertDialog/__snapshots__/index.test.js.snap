// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertDialog /> does not render 1`] = `
<Modal
  active={true}
  from="middle"
  onDismiss={[Function]}
>
  <div
    className="alert"
  >
    yo?
    <div
      className="alert__controls"
    >
      <Button
        className="alert__control"
        color="red"
        onClick={[Function]}
        size="md"
        variant="bare"
      >
        No
      </Button>
      <Button
        color="green"
        onClick={[Function]}
        size="md"
        variant="bare"
      >
        Yes
      </Button>
    </div>
  </div>
</Modal>
`;
